hack/known_types/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"go/importer"
+	"go/types"
+	"io/ioutil"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	if err := newCommand().Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+const (
+	packagePrefix = "k8s.io/api/"
+)
+
+func newCommand() *cobra.Command {
+	var (
+		docsOutputPath string = ""
+	)
+	var command = &cobra.Command{
+		Use:     "go run github.com/argoproj/argo-cd/hack/known_types ALIAS PACKAGE_PATH OUTPUT_PATH",
+		Example: "go run github.com/argoproj/argo-cd/hack/known_types corev1 k8s.io/api/core/v1 corev1_known_types.go",
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) < 3 {
```

Comment:
@jessesuen , @jannfis . Unfortunately, golang reflection does not allow creating instance by type name ( I spent several hours trying to find a way and pretty sure it is not supported ). So I had to use code generation. The generator is used here:

https://github.com/argoproj/argo-cd/pull/3357/files?file-filters%5B%5D=.go#diff-2389d1ede48ad7504a316ba4cf340e9dR16

If need need to support more k8s packages it should be relatively easy to do.